datafusion/common/src/config.rs

```diff
@@ -191,6 +191,8 @@ config_namespace! {
         /// MySQL, PostgreSQL, Hive, SQLite, Snowflake, Redshift, MsSQL, ClickHouse, BigQuery, and Ansi.
         pub dialect: String, default = "generic".to_string()
 
+        /// Should DataFusion prioritize UDF over built-in scalar function in case of function name conflict.
```

Comment:
I actually don't think a config flag is needed -- if a UDF is registered with the same name as a built in, it will not be callable (as you note in the issue / PR description). Therefore I think it is safe to assume that the if a user registers a function with the same name as a built in they want the new function to have precidence

Reply:
Agreed

Reply:
Done.
Thanks for responding so fast, this PR got much more simple.
I'll also edit the PR description - remove the described user facing changes